client/config/config.go

```diff
@@ -0,0 +1,101 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/cosmos/cosmos-sdk/client"
+)
+
+// Default constants
+const (
+	chainID        = ""
+	keyringBackend = "os"
+	output         = "text"
+	node           = "tcp://localhost:26657"
+	broadcastMode  = "sync"
+)
+
+type ClientConfig struct {
+	ChainID        string `mapstructure:"chain-id" json:"chain-id"`
+	KeyringBackend string `mapstructure:"keyring-backend" json:"keyring-backend"`
+	Output         string `mapstructure:"output" json:"output"`
+	Node           string `mapstructure:"node" json:"node"`
+	BroadcastMode  string `mapstructure:"broadcast-mode" json:"broadcast-mode"`
+}
+
+// DefaultClientConfig returns the reference to ClientConfig with default values.
+func DefaultClientConfig() *ClientConfig {
+	return &ClientConfig{chainID, keyringBackend, output, node, broadcastMode}
+}
+
+func (c *ClientConfig) SetChainID(chainID string) {
+	c.ChainID = chainID
+}
+
+func (c *ClientConfig) SetKeyringBackend(keyringBackend string) {
+	c.KeyringBackend = keyringBackend
+}
+
+func (c *ClientConfig) SetOutput(output string) {
+	c.Output = output
+}
+
+func (c *ClientConfig) SetNode(node string) {
+	c.Node = node
+}
+
+func (c *ClientConfig) SetBroadcastMode(broadcastMode string) {
+	c.BroadcastMode = broadcastMode
+}
+
+// ReadFromClientConfig reads values from client.toml file and updates them in client Context
+func ReadFromClientConfig(ctx client.Context) (client.Context, error) {
+	configPath := filepath.Join(ctx.HomeDir, "config")
+	configFilePath := filepath.Join(configPath, "client.toml")
+	conf := DefaultClientConfig()
+
+	// if config.toml file does not exist we create it and write default ClientConfig values into it.
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		if err := ensureConfigPath(configPath); err != nil {
+			return ctx, fmt.Errorf("couldn't make client config: %v", err)
+		}
+
+		configTemplate, err := initConfigTemplate()
```

Comment:
could `initConfigTemplate` and `writeConfigFile` be merged into one function? They are always called together, and we're duplicating the error

Reply:
Yeah. This is good idea. I have merged them into `writeConfigToFile`